packages/query-core/src/query.ts

```diff
@@ -392,6 +395,11 @@ export class Query<
           new Error(`Missing queryFn: '${this.options.queryHash}'`),
         )
       }
+
+      if (this.options.queryFn === skipToken) {
+        return Promise.resolve(undefined)
+      }
```

Comment:
this isn't ideal because a resolved value can and should never be `undefined`. Instead, I guess we should set the `queryFn` itself back to `undefined`, or to a function that throws (maybe in dev mode with a descriptive message?), in `defaultQueryOptions` when we set `enabled: false`.

Usually, the queryFn won't run, but if you pass a `skipToken` and then call `refetch` manually - it would, but that's kind of an invariant

Reply:
Sounds good, on it

Reply:
@TkDodo Are you sure we should set the `queryFn` in `defaultQueryOptions`. I think it should be kept as skipToken in runtime so it will be "easier" to debug and might be also nicer to display in dev tools in future.

If we override it we gain nothing but lose the information

I would prefer to keep the check and instead of 


```suggestion
// Rejects with an error if `queryFn` is set to `skipToken`, indicating that this query should not be executed.
if (this.options.queryFn === skipToken) {
  return Promise.reject(new Error(`Attempted to invoke queryFn when set to skipToken, indicating a configuration error. Query hash: '${this.options.queryHash}'`));
}
```


But I'm open to changing it